src/qforte/ucc/spqe.py

```diff
@@ -439,16 +440,16 @@ def update_ansatz(self):
                     if(self._verbose):
                         print(f"  {Nmu_tup[1]:10}                  {np.real(Nmu_tup[0]):14}")
                     if(Nmu_tup[1] not in self._tops):
-                        self._tops.insert(0,Nmu_tup[1])
+                        self._tops.insert(0,self._excitation_dictionary[Nmu_tup[1]])
```

Comment:
This part of the code was confusing to begin with, and the sheer length of `len(self._pool_obj[self._excitation_dictionary[Nmu_tup[1]]][1].terms()[0][1]) - 1` makes it worse. Might I suggest writing `_, global_op_idx = Nmu_tup` so that you can replace `Nmu_tup[1]`? You can also define `pool_idx = self._excitation_dictionary[global_op_idx]`. I'd also encourage defining `operator_rank = len(self._pool_obj[pool_idx][1].terms()[0][1]) - 1` so that the incrementing of `_nbody_counts` is more transparent. These auxiliary variables should clean up most of the below code, as well.

Good variable names are self-documenting!